### PR TITLE
✨ crd/gen: sort FindKubeKinds

### LINKED ISSUE #675: controller-gen: Deterministic ordering for CRD output manifests (output:stdout)
Every time I run `controller-gen` to generate YAML manifest for my CRDs, I get a different output.

Often, the `CustomResourceDefinition` objects that it spits out and separates with `---` in yaml doc are in different orders. For example if I have CRDs named Type1, Type2, Type3, the order in which they appear in YAML keeps shuffling.

This makes it hard/impossible to write validation/tests that ensure the version-controlled generated artifact (my `crds.yaml`) is up-to-date with what would be generated (hence, prevent drifts + ensure commit consistency).

For posterity, I'm running:

```
go run sigs.k8s.io/controller-tools/cmd/controller-gen@v0.8.0 \
                crd:maxDescLen=0,generateEmbeddedObjectMeta=true \
                paths=./pkg/apis/... output:stdout > ./deployment/crds.yaml
```

I have 5 CRDs so I have 120 permutations, if I run the cmd ~100 times, I definitely see some identical checksums of the output, so I think we just need to order the CRDs by name in the output.

### PR DESCRIPTION
**Pull request by gshirazi**
Fixes #675 by sorting `kubeKinds` in `FindKubeKinds`

### CODE REVIEW

**Comment by gshirazi** [comment]
@ahmetb kindly plz review

**Comment by ahmetb** [comment]
Seems to be working reproducibly and sorting kinds as expected. Thanks.
/lgtm

**Comment by ahmetb** [comment]
needs rebase after #682 is merged.

**Comment by alvaroaleman** [comment]
@gshirazi could you please add a test for this?

**Comment by ahmetb** [comment]
> @gshirazi could you please add a test for this?

at a high level, the Generate() function is largely untested, so I am not sure if it'll be a fair ask to add tests for this without while missing tests for the actual functionality.

but there seems to be some integration tests, so maybe it's possible to write one that captures +output:stdout case.

**Comment by gshirazi** [comment]
I see,

```pkg/crd/gen.go:275:20: cannot use i (variable of type int) as schema.GroupKind value in map index (typecheck)```

I need to rebase this based on the latest master (map not slice). Then add some integration test.

**Comment by gshirazi** [comment]
any objections to reverting https://github.com/kubernetes-sigs/controller-tools/commit/81d0342bf56c92b70f1defef1f2be39c51e1932e ?

**Comment by alvaroaleman** [comment]
> any objections to reverting https://github.com/kubernetes-sigs/controller-tools/commit/81d0342bf56c92b70f1defef1f2be39c51e1932e ?

Yes, it was added to de-duplicate. Please just copy the map keys into a slice and sort that slice rather than reverting anything. Also, we are not going to merge anything here until there are tests.

**Comment by gshirazi** [comment]
@alvaroaleman @ahmetb  added tests. please review, thanks.

**Comment by gshirazi** [comment]
> @alvaroaleman @ahmetb added tests. please review, thanks.

any feedback?

**Comment by ahmetb** [comment]
Sorry, I'm not very initiated in this repository. I will defer to others.

File: pkg/crd/gen.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ import (
 	"fmt"
```
Comment (alvaroaleman) [inline pkg/crd/gen.go:20]: @gshirazi I can't comment on it, but you checked in a binary `cmd/main`. Please remove that, other than that the change looks good to me.
/hold

Comment (gshirazi) [reply pkg/crd/gen.go:20]: done.

**Comment by alvaroaleman** [comment]
/lgtm
/approve
/hold cancel

Thanks!

**Pull request merged**